scripts/pylib/twister/twisterlib/harness.py

```diff
@@ -600,11 +600,16 @@ def _parse_report_file(self, report):
 
 class Gtest(Harness):
     ANSI_ESCAPE = re.compile(r'\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])')
-    TEST_START_PATTERN = r".*\[ RUN      \] (?P<suite_name>[a-zA-Z_][a-zA-Z0-9_]*)\.(?P<test_name>[a-zA-Z_][a-zA-Z0-9_]*)"
-    TEST_PASS_PATTERN = r".*\[       OK \] (?P<suite_name>[a-zA-Z_][a-zA-Z0-9_]*)\.(?P<test_name>[a-zA-Z_][a-zA-Z0-9_]*)"
-    TEST_SKIP_PATTERN = r".*\[ DISABLED \] (?P<suite_name>[a-zA-Z_][a-zA-Z0-9_]*)\.(?P<test_name>[a-zA-Z_][a-zA-Z0-9_]*)"
-    TEST_FAIL_PATTERN = r".*\[  FAILED  \] (?P<suite_name>[a-zA-Z_][a-zA-Z0-9_]*)\.(?P<test_name>[a-zA-Z_][a-zA-Z0-9_]*)"
-    FINISHED_PATTERN = r".*\[==========\] Done running all tests\."
+    _NAME_PATTERN = "[a-zA-Z_][a-zA-Z0-9_]*"
+    _SUITE_TEST_NAME_PATTERN = f"(?P<suite_name>{_NAME_PATTERN})\\.(?P<test_name>{_NAME_PATTERN})"
+    TEST_START_PATTERN = f".*\\[ RUN      \\] {_SUITE_TEST_NAME_PATTERN}"
```

Comment:
can this be further simplified?

Reply:
Can you give an example of what you'd like to see? I felt like this was the right balance of fewer copy pastes and readable.

Reply:
if it is regex as before the blanks can be `\s*` so the 
```
TEST_START_PATTERN = r".*\[\s*RUN\s* \] (?P<suite_name>[a-zA-Z_][a-zA-Z0-9_]*)\.(?P<test_name>[a-zA-Z_][a-zA-Z0-9_]*)"
```

Reply:
But why use a character class when it can/should only be spaces. This shouldn't match tabs or some arbitrary number of spaces.

Reply:
basically, you are right, but the number of blanks is not guaranteed, it would be saft to not count this.

Reply:
We're going to have to disagree there. If we were using an RPC to communicate with the host machine about the status of the tests (which I'd love to get to that point) that would be one thing. But right now we're just scanning the console output as raw text. I feel that we should be as explicit as possible in the regex until we run into a case that fails.